controller/model/posture_check_model_process.go

```diff
@@ -23,12 +23,41 @@ import (
 )
 
 type PostureCheckProcess struct {
+	PostureCheckId  string
 	OperatingSystem string
 	Path            string
 	Hashes          []string
 	Fingerprint     string
 }
 
+func (p *PostureCheckProcess) Evaluate(pd *PostureData) bool {
+	for _, process := range pd.Processes {
+		if process.PostureCheckId == p.PostureCheckId {
+			if process.TimedOut {
+				return false
+			}
+
+			if p.Fingerprint != "" {
+				if p.Fingerprint != process.SignerFingerprint {
+					return false
+				}
+			}
+
+			if len(p.Hashes) > 0 {
+				for _, hash := range p.Hashes {
+					if hash == process.BinaryHash {
+						return true
+					}
+				}
+			} else {
+				return true
```

Comment:
if the PostureCheckProcess has no hashes (p.Hashes > 0) and that's a success - can you move the if len(p.Hashes) <= 0 check outside the for loop and speed up the checking process a smidgeon and make this code here a smidge tighter?

Reply:
I don't think that will improve anything. Readability maybe but that wasn't brought up.